### ISSUE TITLE
Writing to disk does not yet work with httr2

### ISSUE THREAD

**Issue by sckott** [ropensci/vcr#270]
``` r
library(vcr)
library(httr2)

tmpdir <- tempdir()
invisible(vcr_configure(
  dir = file.path(tmpdir, "fixtures"),
  write_disk_path = file.path(tmpdir, "files")
))

my_file <- tempfile(tmpdir = file.path(tmpdir, "files"))

use_cassette(name = "stuff2", {
  request("https://google.com") |> req_perform(path = my_file)
})

readLines(my_file)
#> Warning in file(con, "r"): cannot open file
#> '/var/folders/qt/fzq1m_bj2yb_7b2jz57s9q7c0000gp/T//Rtmp8tCUFj/files/file6ec3313ed849':
#> No such file or directory
#> Error in file(con, "r"): cannot open the connection
```

<sup>Created on 2024-09-04 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by sckott** [comment]
Thanks @Aariq Yep, that's not the correct behavior. Thanks for the comment. Sorry I haven't got around to this yet.

**Comment by sckott** [comment]
There's no way AFAICT to gather the existence or value of `path` with a http request using `httr2::req_perform`. I think it's because we're grabbing the request at this point https://github.com/r-lib/httr2/blob/main/R/req-perform.R#L81-L87 in the function, BEFORE `path` is handled and so there's no `path` information in the request object. `crul` and `httr` have path information in the request so those work. 

@hadley know of any way to capture the `path` value (when used) from a `httr2` request when mocking?

**Comment by hadley** [comment]
Hmmm, why does it matter whether or not you save the body to a file? That's not a property of the request itself.

**Comment by sckott** [comment]
Right, in ruby, python and other lang versions of vcr there's no such thing as writing to disk as part of the data that's written to and read from cassettes - it's always a separate task. But in R's http clients (httr, httr2, crul) we allow users to give a path to write the response to as part of the request incantation. So from a user's perspective it kinda is part of the request. Having said that, maybe you have a good point suggesting maybe removing support for writing to disk in cassettes

**Comment by hadley** [comment]
I think there's a possible argument to be made that saving to a separate file would be a property of the cassette. I can certainly imagine cases where you'd want larger files to be saved separately, but I think that's really about the UI of vcr, not the underlying HTTP library.
